src/systemctl/systemctl-show.c

```
@@ -2488,6 +2492,35 @@ static int show_system_status(sd_bus *bus) {
         return 0;
 }
 
+static int show_manager_varlink_json(void) {
+        _cleanup_(sd_varlink_unrefp) sd_varlink *vl = NULL;
+        _cleanup_(sd_json_variant_unrefp) sd_json_variant *reply = NULL;
+        _cleanup_free_ char *address = NULL;
+        int r;
+
+        if (arg_runtime_scope == RUNTIME_SCOPE_USER) {
+                const char *runtime_dir = secure_getenv("XDG_RUNTIME_DIR");
+                if (!runtime_dir)
+                        return log_error_errno(SYNTHETIC_ERRNO(EUNATCH), "XDG_RUNTIME_DIR not set.");
+
+                address = path_join(runtime_dir, "systemd/io.systemd.Manager");
+        } else
+                address = strdup("/run/systemd/io.systemd.Manager");
+        if (!address)
+                return log_oom();
+
+        r = sd_varlink_connect_address(&vl, address);
+        if (r < 0)
+                return log_error_errno(r, "Failed to connect to %s: %m", address);
+
+        r = varlink_call_and_log(vl, "io.systemd.Manager.Describe", NULL, &reply);
+        if (r < 0)
+                return r;
+
+        sd_json_variant_dump(reply, output_mode_to_json_format_flags(arg_output), stdout, NULL);
+        return 0;
```

Please propagate the failure:
```c
return sd_json_variant_dump(...);
```